fedimint-core/src/encoding/tls.rs

```diff
@@ -10,3 +10,47 @@ impl Encodable for rustls::Certificate {
         self.0.encode_hex::<String>().consensus_encode(writer)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::io::Cursor;
+
+    struct MockCertificate(Vec<u8>);
+
+    impl MockCertificate {
+        fn new(data: Vec<u8>) -> Self {
+            MockCertificate(data)
+        }
+    }
+
+    impl Encodable for MockCertificate {
+        fn consensus_encode<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
+            self.0.encode_hex::<String>().consensus_encode(writer)
+        }
+    }
+
+    #[test]
+    fn test_consensus_encode() {
```

Comment:
What exactly are we testing here? (Please add in a docstring comment as well)

Seems like testing encoding crate in an essence.

Reply:
hi @dpc, I am testing the implementation of the consensus_encode method for the rustls::Certificate type, which is part of the encoding crate.

Reply:
It would make more sense to me to test this by encoding it and then decoding it back to the struct and comparing it against the original struct.

Reply:
Thanks for your input @m1sterc001guy, that makes more sense. I'll definitely implement that approach to further validate the encoding and decoding processes!